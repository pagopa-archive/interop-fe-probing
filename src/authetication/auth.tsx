import awsmobile from './aws-exports'
import { Auth, Amplify } from 'aws-amplify'

Amplify.configure(awsmobile)

async function login(username: string, password: string) {
  try {
    const user = await Auth.signIn(username, password)
    const token = user.signInUserSession.idToken.jwtToken
    const refreshToken = user.signInUserSession.refreshToken.token
    const accessToken = user.signInUserSession.accessToken.jwtToken
    sessionStorage.setItem('token', token)
    sessionStorage.setItem('refreshToken', refreshToken)
    sessionStorage.setItem('accessToken', accessToken)
    return user
  } catch (error) {
    console.log('error signing in', error)
  }
}

async function logout() {
  try {
    await Auth.signOut()
    sessionStorage.clear()
  } catch (error) {
    console.log('error signing out: ', error)
  }
}

export { login, logout }
