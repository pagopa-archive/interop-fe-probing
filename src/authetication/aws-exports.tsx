/* eslint-disable */
// WARNING: DO NOT EDIT. This file is automatically generated by AWS Amplify. It will be overwritten.

const awsmobile = {
  aws_project_region: import.meta.env.VITE_AWS_PROJECT_REGION,
  aws_cognito_identity_pool_id: null,
  aws_cognito_region: import.meta.env.VITE_AWS_COGNITO_REGION,
  aws_user_pools_id: import.meta.env.VITE_AWS_USER_POOL_ID,
  aws_user_pools_web_client_id: import.meta.env.VITE_AWS_USER_POOLS_WEB_CLIENT_ID,
  oauth: {},
  aws_cognito_username_attributes: ['EMAIL'],
  aws_cognito_social_providers: [],
  aws_cognito_signup_attributes: ['EMAIL'],
  aws_cognito_mfa_configuration: 'OFF',
  aws_cognito_mfa_types: [],
  aws_cognito_password_protection_settings: {
    passwordPolicyMinLength: 8,
    passwordPolicyCharacters: [],
  },
  aws_cognito_verification_mechanisms: ['EMAIL'],
}

export default awsmobile
