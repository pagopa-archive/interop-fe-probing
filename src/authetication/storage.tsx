const setStorage = (name: string, value: string) => {
  return new Promise((resolve, reject) => {
    sessionStorage.setItem(name, value);
    return resolve("OK");
  });
};

const deleteStorage = (name: string) => {
  return new Promise((resolve, reject) => {
    return resolve(sessionStorage.removeItem(name));
  });
};

const resetStorage = () => {
  return new Promise((resolve, reject) => {
    return resolve(sessionStorage.clear());
  });
};

const getStorage = (item: string) => {
  return new Promise((resolve, reject) => {
    return resolve(sessionStorage.getItem(item));
  });
};

export { setStorage, deleteStorage, resetStorage, getStorage };
