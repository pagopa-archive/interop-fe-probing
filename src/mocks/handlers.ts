import { rest } from 'msw'

export const handlers = [
  rest.get('http://localhost:3000/eservices', (req, res, ctx) => {
    const mockResponse = {
      content: [
        {
          id: '2',
          eserviceName: 'Anagrafica comunale',
          producerName: 'Comune di Torino',
          state: 'OFFLINE',
          versionNumber: 8,
        },
        {
          id: '1',
          eserviceName: 'Anagrafica comunale',
          producerName: 'Comune di Milano',
          state: 'ONLINE',
          versionNumber: 7,
        },
        {
          id: '3',
          eserviceName: 'Anagrafica comunale',
          producerName: 'Comune di Torino',
          state: 'ONLINE',
          versionNumber: 7,
        },
        {
          id: '10',
          eserviceName: 'Erogazione borse di studio',
          producerName:
            'A.Di.S.U.R.C. Azienda per Il Diritto allo Studio Universitario della Regione Campania',
          state: 'ONLINE',
          versionNumber: 1,
        },
        {
          id: '11',
          eserviceName: 'Funzionamento pannelli solari',
          producerName: 'Comune di Brindisi',
          state: 'ONLINE',
          versionNumber: 1,
        },
        {
          id: '4',
          eserviceName:
            'Posizione centraline ambientali con un nome davvero molto molto molto lungo',
          producerName: 'Comune di Pescara',
          state: 'ONLINE',
          versionNumber: 1,
        },
        {
          id: '5',
          eserviceName: 'Postazioni e-bike',
          producerName: 'Comune di Roma',
          state: 'OFFLINE',
          versionNumber: 1,
        },
        {
          id: '6',
          eserviceName: 'Stato colonnine di ricarica auto elettriche',
          producerName: 'Comune di Bari',
          state: 'OFFLINE',
          versionNumber: 1,
        },
        {
          id: '7',
          eserviceName: 'Stato lavori Olimpiadi 2026',
          producerName: 'Comune di Milano',
          state: 'ONLINE',
          versionNumber: 1,
        },
        {
          id: '8',
          eserviceName: 'Stato opere PGT 2030',
          producerName: 'Comune di Trento',
          state: 'n/d',
          versionNumber: 1,
        },
      ],
      offset: 0,
      limit: 10,
      totalElements: 11,
    }

    return res(ctx.json(mockResponse))
  }),

  rest.get('http://localhost:3000/producers', (req, res, ctx) => {
    const mockResponse = [
      {
        label: 'producer2',
        value: 'producer2',
      },
    ]
    return res(ctx.json(mockResponse))
  }),

  rest.get('http://localhost:3000/eservices/mainData/', (req, res, ctx) => {
    const mockResponse = {
      eserviceName: 'Probing test 3',
      producerName: 'Comune di Milano',
      versionNumber: '7',
      pollingFrequency: 5,
    }
    return res(ctx.json(mockResponse))
  }),

  rest.get('http://localhost:3000/eservices/probingData/', (req, res, ctx) => {
    const mockResponse = {
      probingEnabled: true,
      state: 'offline',
      responseReceived: '2023-04-27T14:55:25Z',
      eserviceActive: false,
    }
    return res(ctx.json(mockResponse))
  }),

  rest.get('http://localhost:3000/telemetryData/eservices/', (req, res, ctx) => {
    const mockResponse = {
      values: [
        {
          responseTime: 640,
          status: 'OK',
          time: '2023-05-16T13:17:01.579Z',
        },
        {
          responseTime: 98,
          status: 'OK',
          time: '2023-05-16T13:33:07.271Z',
        },
        {
          responseTime: 6,
          status: 'OK',
          time: '2023-05-16T13:39:01.371Z',
        },
      ],
      percentages: [
        {
          value: 94.28572,
          status: 'OK',
        },
        {
          value: 5.714286,
          status: 'KO',
        },
      ],
    }
    return res(ctx.json(mockResponse))
  }),
]
