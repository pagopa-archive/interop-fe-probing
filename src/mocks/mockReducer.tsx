import { BrowserRouter as Router } from 'react-router-dom'
import { render } from '@testing-library/react'
import { QueryClient, QueryClientProvider } from '@tanstack/react-query'

const queryClient = new QueryClient()

function reducer(ui: any, { preloadedState, store, ...renderOptions }: any = {}) {
  function Wrapper({ children }: any) {
    return (
      <Router>
        <QueryClientProvider client={queryClient}>{children}</QueryClientProvider>
      </Router>
    )
  }
  return render(ui, { wrapper: Wrapper, ...renderOptions })
}

export { reducer }
