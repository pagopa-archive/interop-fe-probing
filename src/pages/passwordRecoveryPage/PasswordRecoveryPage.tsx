import { PasswordRecoveryForm } from '../../components/forms/passwordRecoveryForm/PasswordRecoveryForm'
import { Typography, Stack, Button, Badge } from '@mui/material'
import { useTranslation } from 'react-i18next'
import { useState } from 'react'
import { useNavigate } from 'react-router-dom'
import MailOutlined from '@mui/icons-material/MailOutlined'
import { IllusEmailValidation } from '@pagopa/mui-italia'

interface SuccessScreenProps {
  title: string
  subtitle: string
  submitButton: string
  onSubmitButton: () => void
}

const SuccessScreen: React.FC<SuccessScreenProps> = ({
  title,
  subtitle,
  submitButton,
  onSubmitButton,
}) => {
  return (
    <Stack
      spacing={4}
      alignItems="center"
      component="main"
      my={9}
      mx={'auto'}
      sx={{ maxWidth: '500px' }}
    >
      <IllusEmailValidation />
      <Typography variant="h3" component="h1">
        {title}
      </Typography>
      <Typography variant="body1" align={'center'}>
        {subtitle}
      </Typography>
      <Button
        sx={{
          backgroundColor: 'primary.main',
          '&:hover': { backgroundColor: 'primary.dark' },
        }}
        size="large"
        variant="contained"
        onClick={() => onSubmitButton()}
      >
        <Typography sx={{ color: 'white' }}> {submitButton}</Typography>
      </Button>
    </Stack>
  )
}

export const PasswordRecoveryPage: React.FC = () => {
  const [recoverySuccess, setRecoverySuccess] = useState(false)

  const { t } = useTranslation(['passwordRecoveryPage'])

  const navigate = useNavigate()

  if (recoverySuccess)
    return (
      <SuccessScreen
        title={t('successScreenTitle', { ns: 'passwordRecoveryPage' })}
        subtitle={t('successScreenSubtitle', { ns: 'passwordRecoveryPage' })}
        submitButton={t('successScreenSubmitButton', { ns: 'passwordRecoveryPage' })}
        onSubmitButton={() => navigate('/monitoraggio')}
      />
    )

  return (
    <Stack spacing={4} alignItems="center" component="main" my={9}>
      <Typography variant="h3" component="h1">
        {t('title', { ns: 'passwordRecoveryPage' })}
      </Typography>
      <PasswordRecoveryForm setRecoverySuccess={setRecoverySuccess} />
    </Stack>
  )
}
