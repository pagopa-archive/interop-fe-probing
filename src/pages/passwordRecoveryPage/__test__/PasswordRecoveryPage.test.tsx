import { describe, expect, test, vi } from 'vitest'
import { PasswordRecoveryPage } from '../PasswordRecoveryPage'
import { renderWithRouterAndQuery } from '../../../mocks/mockWrapper'
import { logRoles, screen, prettyDOM } from '@testing-library/react'
import * as auth from '../../../authetication/auth'
import userEvent from '@testing-library/user-event'
import * as router from 'react-router'

describe('PasswordRecoveryPage', () => {
  test('render component', () => {
    const { container } = renderWithRouterAndQuery(<PasswordRecoveryPage />)
    expect(container).toMatchSnapshot()
  })

  test('fill field and click submit simulating success', async () => {
    const spyAuth = vi.spyOn(auth, 'passwordRecovery')
    spyAuth.mockImplementation(() => Promise.resolve())

    const { container } = renderWithRouterAndQuery(<PasswordRecoveryPage />)
    const submitButton = screen.getByRole('button')
    const emailField = screen.getByRole('textbox')
    const user = userEvent.setup()
    await user.type(emailField, 'test@test.com')
    await user.click(submitButton)
    expect(spyAuth).toBeCalled()
    expect(container).not.toMatchSnapshot()
    expect(screen.getByText(/successScreenTitle/i))
    expect(screen.getByText(/successScreenSubtitle/i))
  })
})
