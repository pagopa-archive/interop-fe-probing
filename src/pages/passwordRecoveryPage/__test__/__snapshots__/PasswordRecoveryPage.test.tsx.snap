// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`PasswordRecoveryPage > fill field and click submit simulating success 1`] = `
<div>
  <main
    class="MuiStack-root css-a7mbvr-MuiStack-root"
  >
    <svg
      aria-hidden="true"
      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-11e46a8-MuiSvgIcon-root"
      focusable="false"
      role="img"
      viewBox="0 0 120 120"
    >
      <path
        d="M37.066 87.421a2.32 2.32 0 0 0-2.324 2.314 2.32 2.32 0 0 0 2.324 2.315 2.32 2.32 0 0 0 2.325-2.314 2.32 2.32 0 0 0-2.325-2.315Z"
      />
      <path
        clip-rule="evenodd"
        d="M78.105 36.512C79.265 26.114 88.145 18 98.895 18c11.534 0 20.919 9.343 20.919 20.826 0 10.702-8.15 19.542-18.595 20.698v34.84c0 3.825-3.123 6.942-6.973 6.942H7.779c-3.879 0-6.973-3.145-6.973-6.942v-50.91c0-3.854 3.15-6.942 6.973-6.942h70.326Zm-41.32 31.224L5.456 43.583v50.39l31.33-26.237Zm3.76 2.899 6.127 4.72a6.966 6.966 0 0 0 4.34 1.523 6.97 6.97 0 0 0 4.333-1.516l6.922-5.258 31.458 26.574H9.445l31.1-26.043Zm43.539-17.113L66.049 67.23 96.57 93.01V59.525a20.823 20.823 0 0 1-12.486-6.002ZM52.506 71.698a.467.467 0 0 0-.017.013l-.027.02c-.854.68-2.046.68-2.9 0l-.03-.023C49.5 71.683 9.904 41.158 9.879 41.14H78.11a20.827 20.827 0 0 0 3.08 8.759L52.506 71.698Zm33.688-22.761a16.23 16.23 0 0 0 12.7 6.088c8.972 0 16.271-7.267 16.271-16.199 0-8.931-7.299-16.198-16.27-16.198-8.972 0-16.271 7.267-16.271 16.198 0 3.718 1.235 7.214 3.57 10.11Z"
        fill-rule="evenodd"
      />
      <path
        d="m106.549 32.562-9.979 9.934-5.33-5.306a2.33 2.33 0 0 0-3.286 0 2.306 2.306 0 0 0 0 3.273l6.973 6.942a2.331 2.331 0 0 0 3.287 0l11.622-11.57a2.306 2.306 0 0 0 0-3.273 2.331 2.331 0 0 0-3.287 0ZM66.353 87.421h-19.99a2.319 2.319 0 0 0-2.324 2.314 2.319 2.319 0 0 0 2.325 2.315h19.99a2.32 2.32 0 0 0 2.324-2.314 2.319 2.319 0 0 0-2.325-2.315Z"
      />
      <title>
        Email verificata correttamente
      </title>
    </svg>
    <h1
      class="MuiTypography-root MuiTypography-h3 css-gepadz-MuiTypography-root"
    >
      successScreenTitle
    </h1>
    <p
      class="MuiTypography-root MuiTypography-body1 MuiTypography-alignCenter css-1uwgr7b-MuiTypography-root"
    >
      successScreenSubtitle
    </p>
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeLarge MuiButton-containedSizeLarge MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeLarge MuiButton-containedSizeLarge css-w4y956-MuiButtonBase-root-MuiButton-root"
      tabindex="0"
      type="button"
    >
      <p
        class="MuiTypography-root MuiTypography-body1 css-jb1ygz-MuiTypography-root"
      >
         
        successScreenSubmitButton
      </p>
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
  </main>
</div>
`;

exports[`PasswordRecoveryPage > render component 1`] = `
<div>
  <main
    class="MuiStack-root css-1bw9v5o-MuiStack-root"
  >
    <h1
      class="MuiTypography-root MuiTypography-h3 css-gepadz-MuiTypography-root"
    >
      title
    </h1>
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation16 MuiCard-root css-9cvwhb-MuiPaper-root-MuiCard-root"
    >
      <form>
        <div
          class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-direction-xs-column css-ljy4ha-MuiGrid-root"
        >
          <div
            class="MuiGrid-root MuiGrid-container MuiGrid-item css-1qy0967-MuiGrid-root"
          >
            <div
              class="MuiGrid-root MuiGrid-container MuiGrid-item css-1f064cs-MuiGrid-root"
            >
              <div
                class="MuiFormControl-root MuiTextField-root css-p21bt9-MuiFormControl-root-MuiTextField-root"
              >
                <label
                  class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-colorPrimary Mui-required MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
                  data-shrink="false"
                  for="email"
                  id="email-label"
                >
                  Email
                  <span
                    aria-hidden="true"
                    class="MuiFormLabel-asterisk MuiInputLabel-asterisk css-wgai2y-MuiFormLabel-asterisk"
                  >
                     
                    *
                  </span>
                </label>
                <div
                  class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
                >
                  <input
                    aria-invalid="false"
                    class="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                    id="email"
                    required=""
                    type="email"
                    value=""
                  />
                  <fieldset
                    aria-hidden="true"
                    class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                  >
                    <legend
                      class="css-yjsfm1"
                    >
                      <span>
                        Email
                         
                        *
                      </span>
                    </legend>
                  </fieldset>
                </div>
              </div>
            </div>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item css-131xi47-MuiGrid-root"
          >
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeLarge MuiButton-containedSizeLarge Mui-disabled MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeLarge MuiButton-containedSizeLarge css-w4y956-MuiButtonBase-root-MuiButton-root"
              disabled=""
              tabindex="-1"
              type="submit"
            >
              <p
                class="MuiTypography-root MuiTypography-body1 css-jb1ygz-MuiTypography-root"
              >
                 
                submit
              </p>
            </button>
          </div>
        </div>
      </form>
    </div>
  </main>
</div>
`;
